lib/src/sdk_env.dart

```diff
@@ -436,7 +436,9 @@ class ToolEnvironment {
     final parsed = yamlToJson(original);
     parsed.remove('dev_dependencies');
     parsed.remove('dependency_overrides');
-    parsed['dev_dependencies'] = {'pedantic': '^1.0.0'};
+    if (parsed['name'] != 'pedantic') {
```

Comment:
Maybe add a comment explaining that this is to support analysis of pedantic?

I'm a little bit wondering why we do it this way... rather than feeding the analyzer rules using a command line argument..  but that seems like cleanup for another day.

Reply:
> I'm a little bit wondering why we do it this way... rather than feeding the analyzer rules using a command line argument.. but that seems like cleanup for another day.

We do feed the `analysis-options.yaml` as a command line parameter, but it needs to reference to pedantic's own ruleset, and (AFAICT) that can be only done when `package:pedantic` is added as a dependency.

Reply:
In addition to the above, we should also check if `pedantic` is already in the `dependencies` block.

Reply:
PTAL